website/blog/2022-08-22-unit-testing-dbt-package.md

```diff
@@ -0,0 +1,175 @@
+---
+title: "An introduction to unit testing your dbt Packages"
+description: "Traditionally, integration tests have been the primary strategy for testing dbt Packages. In this post, Yu Ishikawa walks us through adding in unit testing as well."
+slug: unit-testing-dbt-packages
+authors: [yu_ishikawa]
+tags: [dbt tutorials]
+hide_table_of_contents: false
+
+date: 2022-08-25
+is_featured: true
+---
+
+
+It’s important to be able to test any dbt Project, but it’s even more important to make sure you have robust testing if you are developing a [dbt Package](https://docs.getdbt.com/docs/building-a-dbt-project/package-management).
+
+I love dbt packages, because it makes it easy to extend dbt’s functionality and create reusable analytics resources. Even better, we can find and share dbt packages which others developed, finding great packages in [dbt hub](https://hub.getdbt.com/). However, it is a bit difficult to develop complicated dbt macros, because dbt on top of [Jinja2](https://palletsprojects.com/p/jinja/) is lacking some of the functionality you’d expect for software development - like unit testing.
```

Comment:
```suggestion
I love dbt packages, because they make it easy to extend dbt’s functionality and create reusable analytics resources. Even better, we can find and share dbt packages which others developed, finding great packages in [dbt Package Hub](https://hub.getdbt.com/). However, it is a bit difficult to develop complicated dbt macros, because dbt on top of [Jinja2](https://palletsprojects.com/p/jinja/) is lacking some of the functionality you’d expect for software development—like unit testing.
```